src/build_scripts/buildlib.pio

```diff
@@ -137,21 +131,25 @@ def _main_func(description):
             valid_values += ",pnetcdf"
         if netcdf4_parallel_found:
             valid_values += ",netcdf4p,netcdf4c"
-        logger.warning("Updating valid_values for PIO_TYPENAME: %s", valid_values)
+        logger.warning("Updating valid_values for PIO_TYPENAME: {}".format(valid_values))
 
         case.set_valid_values("PIO_TYPENAME",valid_values)
         # nothing means use the general default
         valid_values += ",nothing"
 
         for comp in case.get_values("COMP_CLASSES"):
-            comp_pio_typename = "%s_PIO_TYPENAME"%comp
+            comp_pio_typename = "{}_PIO_TYPENAME".format(comp)
             case.set_valid_values(comp_pio_typename,valid_values)
             current_value = case.get_value(comp_pio_typename)
             if current_value not in valid_values:
-                logger.warning("Resetting PIO_TYPENAME to netcdf for component %s"%comp)
+                logger.warning("Resetting PIO_TYPENAME to netcdf for component {}".format(comp))
                 case.set_value(comp_pio_typename,"netcdf")
 
 
-
 if (__name__ == "__main__"):
-    _main_func(__doc__)
+    if ("--test" in sys.argv):
+        test_results = doctest.testmod(verbose=True)
+        sys.exit(1 if test_results.failed > 0 else 0)
+
+    mbldroot, minstallpath, mcaseroot = parse_command_line(sys.argv, __doc__)
```

Comment:
I think you still want to use a _main_func approach here. All these variables are being inserted into the global namespace.